test/connectivity/README.md

```diff
@@ -69,3 +72,106 @@ Network offline/online should not affect this test result since Edge hub stores
+
+##### Test Mechanism: 
+	1. The Direct Method Cloud Sender invokes a test direct method to the IoT Hub
+	2. The IoT Hub forwards the direct method to Edge Hub
+	3. The Edge Hub forwards the direct method to Direct Method Receiver
+	4. The Direct Method Receiver notifies Test Analyzer regarding the received direct method
+	5. The Direct Method Receiver responds to Edge Hub's direct method invocation
+	6. The Edge Hub  responds to IoT Hub direct method invocation 
+	7. The Direct Method Cloud Sender receives a response from IoT Hub
+	8. The Direct Method Cloud Sender notifies Test Analyzer regarding its direct method response
+Test Analyzer compile a report from Direct Method Cloud Sender, Direct Method Receiver, Network Controller and determine if the test case is passed or failed.
+
+#### Cloud to EdgeAgent Direct Method Test (PING)
+
+![Connectivity Test Diagram](./images/ConnectivityTest_Ping_DirectMethod.png)
+
+##### Test Specific Component:
+	1. Direct Method (Cloud) Sender is an Azure service client that is wrapped in a container module that is deployed into a test agent.  It is responsible for sending a "HelloWorldMethod" direct method to the EdgeAgent module and report the direct method status to Test Result Coordinator
+
+##### Test Mechanism: 
+	1. The Direct Method Cloud Sender invokes a test direct method to the IoT Hub
+	2. The IoT Hub forwards the direct method to Edge Agent
+	3. The Edge Agent has a pre-implemented "ping" method with a response.
+	4. The Direct Method Cloud Sender receives a response from IoT Hub
+	5. The Direct Method Cloud Sender notifies Test Analyzer regarding its direct method response
+Test Analyzer compile a report from Direct Method Cloud Sender, Direct Method Receiver, Network Controller and determine if the test case is passed or failed. 
+
+#### Cloud to Device Message Test
+
+![Connectivity Test Diagram](./images/ConnectivityTest_C2D_Message.png)
+
+The goal of the test scheme is to verify the recoverability of the cloud to device messaging once the device goes offline. 
+##### Test Specific Components:
+	1. Message Sender is a Azure service client that is deployed into a test agent as a container module. The responsibility of the module is to use the Azure API to trigger IoT Hub to send a message to device along with reporting to Test Result Coordinator once a message is sent. 
+	2. Message Receiver is a device client that is wrapped in a module container for a deployment into a test agent. The device module behaves as a target receiving end of the message sending from Message Sender and is responsible for reporting to the Test Result Coordinator once a message is received.
+
+##### Test Implementation Assumption:
+	1. The cloud takes less than 15 minutes to forward all of messages from the cloud to device when the Edge Hub is online.
+	2. Each cycle of Network Controller online-offline period must be longer than the Edge Hub recovering period
```

Comment:
what happens if network online-offline period is shorter than Edge Hub recovering period?